### DISCUSSION TITLE
How can I use C# like javascript

### DISCUSSION THREAD

**Discussion by Xyncgas** [dotnet/runtime#81330] [Q&A]
Now we can run dotnet in browser with our Blazor Webassembly projects, it's tempting to fetch a C# file like we fetch a javascript file, then running the codes

the reason this said file wasn't in a better form to be consumed was the benefit it gives for human to read the codes and understand what it's saying. Meanwhile a program written with the C# language is simply text before it's compiled, making this format very very portable, one very good example is dotnet interactive

I am possitive there is going to be CORS issue etc etc but can we do this

**Comment by huoyaoyuan** [comment]
If you can make everyone install a dotnet runtime like a browser?

**Reply by vitek-karas** [reply]
Maybe a better way to state the problem is download size. The current Blazor WASM projects rely on trimming to reduce the download size necessary to run the code. If nothing else, to enable running random C# files, one would need to download untrimmed version of the .NET runtime and libraries which is very big. Additionally the app would need to also download the entire compiler which is also very big. I'm sure there are also technical limitation currently in place which would make it difficult to support loading random code. So it's probably all solvable, but the downsides are significant.

**Reply by Xyncgas** [reply]
If we bring MAUI.Blazor to the windows browser, we can run any random C# source code, then isolating their effect from other user instances and other processes running would be doable..

Then you have source files referencing stuffs that's not available and there's nuget package and a lot to bootstrap given the tool chain existed before this sudden impulse of mien to run C# source code like they are javascript

Although with star link internet it's not that bad to download everything I guess

**Reply by Clockwork-Muse** [reply]
Star Link, for a number of reasons, is not going to be a panacea.
It's also not relevant as a solution to the real issues with deploying websites/webapps; namely, you want to distribute the smallest, most contained, files you can, so the site loads faster, and without external dependency pulls.

**Comment by Clockwork-Muse** [comment]
.... The whole point of [WebAssembly (WASM)](https://en.wikipedia.org/wiki/WebAssembly) is that you would distribute a language-neutral binary format, especially because sending text is huge compared to binary.  So the trend is to move to a future where _no_ source code is distributed.
